src/modules/sync/utils/contentTypeHelpers.ts

```diff
@@ -0,0 +1,77 @@
+import { ContentTypeContracts, ContentTypeElements, ContentTypeSnippetContracts } from "@kontent-ai/management-sdk";
+import { PortableTextInternalLink, PortableTextObject, traversePortableText } from "@kontent-ai/rich-text-resolver";
+
+export const getGuidelinesElements = (
+  types: ContentTypeContracts.IContentTypeContract[] | ContentTypeSnippetContracts.IContentTypeSnippetContract[],
+) =>
+  types.flatMap(type =>
+    type.elements.filter(element => element.type === "guidelines")
+  ) as unknown as ContentTypeElements.IGuidelinesElement[];
+
+export const getAssetElements = (
+  types: ContentTypeContracts.IContentTypeContract[] | ContentTypeSnippetContracts.IContentTypeSnippetContract[],
+) =>
+  types.flatMap(type =>
+    type.elements.filter(element => element.type === "asset")
+  ) as ContentTypeElements.IAssetElement[];
+
+export const getLinkedItemsElements = (
+  types: ContentTypeContracts.IContentTypeContract[] | ContentTypeSnippetContracts.IContentTypeSnippetContract[],
+) =>
+  types.flatMap(type =>
+    type.elements.filter(element => element.type === "modular_content")
+  ) as unknown as ContentTypeElements.ILinkedItemsElement[];
+
+export const getRequiredAssetsIds = (
+  assetElements: ContentTypeElements.IAssetElement[],
+  portableTextGuidelines: ReadonlyArray<PortableTextObject[]>,
+) => {
+  const result = new Set<string>();
+
+  assetElements.forEach(asset => {
+    asset.default?.global.value.forEach(a => {
+      result.add(a.id as string);
+    });
+  });
+
+  const traverseCallback = (object: PortableTextObject) => {
+    if (object._type === "image") {
+      result.add(object.asset._ref);
+    }
+
+    return object;
+  };
+
+  portableTextGuidelines.map(guideline => guideline.map(g => traversePortableText(g, traverseCallback)));
```

Comment:
well the documentation of the callback says @param callback
A callback function invoked for each node in the Portable Text structure. It can return a modified version of the node or undefined if no modifications are to be made. I want to traverse the the whole tree and find assetIds and itemIds :D I believe I can traverse it by myself but why would i bother when there is a function that does that for me? I am not sure if you want me to use some reduce with recursion (or one of them by itself).. This PR you are speaking in riddles :D

Reply:
or you mean transform to json? :)